UI/src/view/point_asset/trafficLightForm.js

```diff
@@ -0,0 +1,339 @@
+                selectedAsset.set({toBeDeleted: eventTarget.prop('checked')});
+            });
+
+            rootElement.find('.suggested-checkbox').on('change', function (event) {
+                var eventTarget = $(event.currentTarget);
+                var propertyGroupedId = eventTarget.data('grouped-id');
+                var propertyPublicId = _.head(_.split(eventTarget.attr('id'), '-'));
+                selectedAsset.setPropertyByGroupedIdAndPublicId(propertyGroupedId, propertyPublicId, +eventTarget.prop('checked'));
+
+                if (id && !selectedAsset.getWasOldAsset()) {
+                    me.switchSuggestedValue(true);
+                    rootElement.find('.suggested-checkbox').prop('checked', false);
+                }
+            });
+
+            rootElement.find('.editable').not('.suggestion-box').on('change', function () {
+                if (id && !selectedAsset.getWasOldAsset()) {
+                    me.switchSuggestedValue(true);
+                    rootElement.find('.suggested-checkbox').prop('checked', false);
+                    _.forEach(rootElement.find('.suggested-checkbox'), function (element) {
+                        var targetElement = $(element);
+                        var propertyGroupedId = targetElement.data('grouped-id');
+                        var propertyPublicId = _.head(_.split(targetElement.attr('id'), '-'));
+                        selectedAsset.setPropertyByGroupedIdAndPublicId(propertyGroupedId, propertyPublicId, 0);
+                    });
+                }
+            });
+
+            rootElement.find('.point-asset button.save').on('click', function () {
+                selectedAsset.save();
+            });
+
+            rootElement.find('.point-asset button.cancel').on('click', function () {
+                me.switchSuggestedValue(false);
+                selectedAsset.cancel();
+            });
+        };
+
+        this.selectableAssetButton = function(selectedAsset, assetNumber, groupedId) {
+            var assetButton = $('<span groupedId="' + groupedId + '" class="selectable">' + assetNumber + '</span>');
+            if (selectedAsset.getSelectedGroupedId() == groupedId){
+                assetButton.addClass('marker-highlight');
+            } else {
+                assetButton.addClass('marker');
+            }
+
+            return assetButton;
+        };
+
+        this.renderValidityDirection = function (id) {
+            return '' +
+              '    <div class="form-group editable form-directional-traffic-sign edit-only">' +
+              '      <label class="control-label">Vaikutussuunta</label>' +
+              '      <button groupedId="'+id+'" class="change-validity-direction form-control btn btn-secondary btn-block">Vaihda suuntaa</button>' +
+              '    </div>';
+        };
+
+        this.renderValueElement = function(selectedAsset, collection, authorizationPolicy) {
+            var toRet = $();
+            var i = 0;
```

Comment:
Can you give more suggestive names in these fields?

Reply:
names modified